src/message.rs

```diff
@@ -132,6 +132,44 @@ impl MsgId {
         Ok(())
     }
 
+    /// Returns autodelete timer value for the message.
+    pub(crate) async fn autodelete_timer(
+        self,
+        context: &Context,
+    ) -> crate::sql::Result<Option<i64>> {
+        let res = match context
+            .sql
+            .query_get_value_result(
+                "SELECT autodelete_timer FROM msgs WHERE id=?",
+                paramsv![self],
+            )
+            .await?
+        {
+            None | Some(0) => None,
+            Some(timer) => Some(timer),
+        };
+        Ok(res)
+    }
+
+    /// Starts autodelete timer for the message if it is not started yet.
+    pub(crate) async fn start_autodelete_timer(self, context: &Context) -> crate::sql::Result<()> {
+        if let Some(autodelete_timer) = self.autodelete_timer(context).await? {
+            let autodelete_timestamp = time() + autodelete_timer;
+
+            context
+                .sql
+                .execute(
+                    "UPDATE msgs SET autodelete_timestamp = ? \
+                WHERE (autodelete_timestamp == 0 OR autodelete_timestamp > ?) \
+                AND id = ?",
+                    paramsv![autodelete_timestamp, autodelete_timestamp, self],
+                )
+                .await?;
+            schedule_autodelete_task(context).await;
```

Comment:
I realise I'm asking the same thing twice, but this is confusing because all this does is schedule an `MsgsChanged` event to be emitted.  This seems to imply more, maybe it even expects more?

Reply:
Emitting a `MsgsChanged` causes chatlist to be reloaded, and we remove outdated messages before loading the chatlist (including last message in the preview).

Reply:
`MsgsChanged` signals to the UI that some messages may have been deleted, and it should reload chatlist and messagelist it displays.

Reply:
So we rely on the UI to reload the chatlist in order to delete messages?  I find that rather scary, this isn't even explained on the API.

Can't we do this the other way round and schedule actual deletion, when we delete it tell the UI the messages have changed?

Reply:
Chat and chatlist reload have to call deletion anyway, because otherwise there is a chance that we open the app and it shows lots of old messages that get deleted after a second.

If we do it this way, we cause a chain of deletion -> reload -> deletion, where the second deletion is an useless request to the database that does not delete anything unless reload call was very slow.

I prefer to think of the functions that load the chat/chatlist as always returning the correct result, and these tasks are needed to remind the UI to reload the data when it is about to change. Maybe we shouldn't call the task "autodelete task", because it does not actually delete anything. It is a "update view task" or something like that.

Reply:
Ok, I was thinking for a while that I wasn't happy with this but I think this is fine.  There is no way to ensure deletion at a given time anyway because the app might be killed.  So this probably is the right design.

My main remaining concern is the user expectation.  Someone who has this set to say 60s, looks at an ephemeral message for 5s and closes the app.  Now before they can open the app again e.g. their phone is seized and the data is still on there for someone who looks directly at the database.  How do we make users aware of this constraint?  Or is this too theoretical?  /cc @r10s @hpk42 as they might have opinions